### PR TITLE
Fix home constant update

### PR DESCRIPTION
**Pull request by joaquimrocha**
This PR fixes the useWarningsList and useClustersVersion so they do not update whenever there's new data from the server but instead when the data differs from what we returned the last time. (Same goal as #2514)

how to test
- [ ] You can either use dev tools or add a console.log into the HomePure component to see how often it is rendered. With this change it should only rerender when something changes.
- [x] stop a cluster, and start a cluster... the home page should update to show the correct cluster active status

### CODE REVIEW

File: frontend/src/lib/k8s/event.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { useMemo } from 'react';
+import _ from 'lodash';
```
Comment (illume) [inline frontend/src/lib/k8s/event.ts:1]: nit: It's better to import just what you need, for shaking.

File: frontend/src/lib/k8s/event.ts [hunk 2]
```diff
@@ -238,7 +239,19 @@ class Event extends KubeObject<KubeEvent> {
       options?.queryParams ?? {}
     );
 
-    return this.useListForClusters(clusters, { queryParams: queryParameters });
+    const newWarningsList = this.useListForClusters(clusters, { queryParams: queryParameters });
```
Comment (illume) [inline frontend/src/lib/k8s/event.ts:242]: For later... we should probably investigate doing this change inside of "useListForClusters"? Or I thought this had a caching layer built in?

**Review by illume — COMMENTED** [review]
Hrmm. This doesn't fix the issue of the home rendering several times every five seconds.

image

File: frontend/src/lib/k8s/index.ts [hunk 3]
```diff
@@ -297,8 +297,11 @@ export function useClustersVersion(clusters: Cluster[]) {
   const lastUpdateRef = React.useRef(0);
 
   React.useEffect(() => {
-    const newClusterNames = Object.values(clusters).map(c => c.name);
-    if (_.isEqual(newClusterNames, clusterNames)) {
+    const newClusterNames = Object.values(clusters)
```
Comment (illume) [inline frontend/src/lib/k8s/index.ts:300]: Can you please add a comment for why this is here?

File: frontend/src/components/App/Home/index.tsx [hunk 4]
```diff
@@ -174,84 +175,125 @@ interface HomeComponentProps {
...
-              getValue: () => '',
-              cellProps: {
-                align: 'right',
+  React.useEffect(() => {
+    setWarningLabels(currentWarningLabels => {
+      const newWarningLabels: { [cluster: string]: string } = {};
+      for (const cluster of clusterNames) {
+        newWarningLabels[cluster] = renderWarningsText(warningsMap, cluster);
+      }
+      if (!_.isEqual(newWarningLabels, currentWarningLabels)) {
```
Comment (sniok) [inline frontend/src/components/App/Home/index.tsx:202]: _ import is missing, replace with `if(isEqual(...`

File: frontend/src/lib/k8s/event.ts [hunk 5]
```diff
@@ -238,7 +239,19 @@ class Event extends KubeObject<KubeEvent> {
...
+    const [warningList, setWarningList] = React.useState<typeof newWarningsList>(newWarningsList);
+
+    // Only update the warnings if they actually differ
+    React.useEffect(() => {
+      if (_.isEqual(warningList, newWarningsList)) {
```
Comment (sniok) [inline frontend/src/lib/k8s/event.ts:247]: this might be slow when comparing a lot of warnings, but we can profile and look into it later

**Review by sniok — APPROVED** [review]

**Pull request merged**